### ISSUE TITLE
W6-Submission

### ISSUE THREAD

**Issue by Nedu** [Nedu/instagram#1]
My app is complete, please review. /cc @codepathreview

**Comment by avo1** [comment]
Hi @Nedu , how can I upload an image? I don't see you show that in your gif, and when I clone your project, I also can't do it. Tapping on `Upload` didn't bring me to anywhere. 

Another bug is that when I left everything blank and tap on `Sign up`, it still lets me it. Shouldn't be like that?

It looks like you are missing some features from the required user stories. We will need to have this completed and updated within **24hr** from this point in order for you to receive credit for the assignment.

Dave
CodePath

**Comment by Nedu** [comment]
Hi, I have completed it and updated the gif.

**Comment by avo1** [comment]
Is this your latest code? I don't see any picture in the home page after you uploading new image?

User can view the last 20 posts submitted to "Instagram"

**Comment by Nedu** [comment]
hi, yes this is my latest code. the image doesnt show in my gif because of my internet's connection but the print statement shows that it is successfully retrieving the images.

**Comment by avo1** [comment]
No, it's not because of connection. You can retrieve the images but you are not able to show it.
1. Check the autoLayout as suggested below (why do you want to use the aspect ratio?)

<img width="427" alt="screen shot 2016-03-10 at 12 30 56 am" src="https://cloud.githubusercontent.com/assets/10482065/13644292/8cc4bf2e-e657-11e5-9f54-3cfc496086b0.png">
1. Look at the `pictureTableViewCell.swift` This is where you got problem.
- Why do you want to `as!` and then `String?` 
- What is the type of `upload["image"]` Have you tried to print it? Always nil. You need to understand the structure of the "thing" returned by API to be able to extract correct piece of info you need.

<img width="639" alt="screen shot 2016-03-10 at 12 34 17 am" src="https://cloud.githubusercontent.com/assets/10482065/13644341/dcdabda6-e657-11e5-842a-e7c866e00eac.png">

Modify that code and you will be able to see the images on your home page. This is screen captured from you app after I fixed it.

<img width="377" alt="screen shot 2016-03-10 at 12 39 38 am" src="https://cloud.githubusercontent.com/assets/10482065/13644486/9b03db78-e658-11e5-9d1f-2608640b2e4c.png">

Please update your code within **24hr**.

Dave
CodePath

**Comment by Nedu** [comment]
hi, I have fixed the code and updated the gif

**Comment by avo1** [comment]
:+1: I saw the updated code. Congratulations on finishing the final assignment! :tada: It's impressive to realize how far you've come in the first half of the class, going from building a simple tip calculator to a full Instagram app with your own backend.

We hope you enjoyed the final assignment. Parse is a great service that makes building a backend app much easier. We have a detailed [Project 6 Feedback Guide](http://courses.codepath.com/snippets/ios_university/feedback_guides/project_6_feedback.md) which covers the best practices for implementing this assignment. Read through the feedback guide point-by-point to determine ways you might be able to improve your submission.

If you have any particular questions about the assignment or the feedback, feel free to reply here or email us at universitysupport@codepath.com.

From here on out there are no more assignments as the focus shifts to the final group project. You'll be working over the next several weeks to build your app idea with your team and then finishing the class with a demo day!

Dave
CodePath
